libs/fgdbg/include/fgdbg/Session.h

```diff
@@ -17,15 +29,29 @@ class Session {
 
 public:
     Session(const std::string& name,
-            std::shared_ptr<DebugServer> server) : name{ name }, server{ server } {}
+            std::shared_ptr<DebugServer> server) : name{ name }, server{ std::move(server) } {}
```

Comment:
This `std::move` here is confusing and probably unnecessary. The `shared_ptr` is passed by value into the `Session` constructor, so it has already been copied. There's no need to `std::move` it.

Reply:
I'm not great with move-semantics, CLION likes `std::move`:
> Clang-Tidy: Parameter 'server' is passed by value and only copied once; consider moving it to avoid unnecessary copies

Reply:
I think the `std::move` does give a very small performance win here, but in general I tend to avoid moving things unless I have a strong reason as it can lead to subtle bugs. For example, if you tried to access the `server` parameter after the move, it would be null. I'm fine either way though.

Reply:
Thanks for the clarification. I agree moving here is unsafe and copying a pointer is cheap. Done